code/modules/antagonists/heretic/structures/mawed_crucible.dm

```diff
@@ -99,6 +115,16 @@
 		set_anchored(!anchored)
 		balloon_alert(user, "[anchored ? "":"un"]anchored")
 		return ITEM_INTERACT_SUCCESS
+	if(istype(tool, /obj/item/reagent_containers/cup/beaker/eldritch))
+		var/obj/item/reagent_containers/cup/beaker/eldritch/to_fill = tool
+		if(to_fill.reagents.total_volume >= to_fill.reagents.maximum_volume)
+			balloon_alert(user, "flask too full")
+			return ITEM_INTERACT_SUCCESS
+		to_fill.reagents.add_reagent(/datum/reagent/eldritch, 50)
+		do_item_attack_animation(src, used_item = tool)
+		current_mass--
```

Comment:
Feel like this should use all the mass, since it's equivalent to "making a new potion"